pkg/providers/cloudstack/decoder/decoder_test.go

```diff
@@ -42,130 +28,126 @@ func (tctx *testContext) restoreContext() {
+	}{
+		{
+			name:       "Valid config",
+			configFile: "../testdata/cloudstack_config_valid.ini",
+			wantErr:    false,
+			wantConfig: &decoder.CloudStackExecConfig{
+				Instances: []decoder.CloudStackInstanceConfig{
+					{
+						Name:          "Instance1",
+						ApiKey:        "test-key1",
+						SecretKey:     "test-secret1",
+						ManagementUrl: "http://127.16.0.1:8080/client/api",
+					},
+				},
+				VerifySsl: "false",
+				Timeout:   "",
+			},
+		},
+		{
+			name:       "Multiple instances config",
+			configFile: "../testdata/cloudstack_config_multiple_instances.ini",
+			wantErr:    false,
+			wantConfig: &decoder.CloudStackExecConfig{
+				Instances: []decoder.CloudStackInstanceConfig{
+					{
+						Name:          "Instance1",
+						ApiKey:        "test-key1",
+						SecretKey:     "test-secret1",
+						ManagementUrl: "http://127.16.0.1:8080/client/api",
+					},
+					{
+						Name:          "Instance2",
+						ApiKey:        "test-key2",
+						SecretKey:     "test-secret2",
+						ManagementUrl: "http://127.16.0.2:8080/client/api",
+					},
+				},
+				VerifySsl: "false",
+				Timeout:   "",
+			},
+		},
+		{
+			name:       "Missing apikey",
+			configFile: "../testdata/cloudstack_config_missing_apikey.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "Missing secretkey",
+			configFile: "../testdata/cloudstack_config_missing_secretkey.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "Missing apiurl",
+			configFile: "../testdata/cloudstack_config_missing_apiurl.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "Missing verifyssl",
+			configFile: "../testdata/cloudstack_config_missing_verifyssl.ini",
+			wantErr:    false,
+			wantConfig: &decoder.CloudStackExecConfig{
+				Instances: []decoder.CloudStackInstanceConfig{
+					{
+						Name:          "Instance1",
+						ApiKey:        "test-key1",
+						SecretKey:     "test-secret1",
+						ManagementUrl: "http://127.16.0.1:8080/client/api",
+					},
+				},
+				VerifySsl: "true",
+				Timeout:   "",
+			},
+		},
+		{
+			name:       "Invalid verifyssl",
+			configFile: "../testdata/cloudstack_config_missing_global_section.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "Missing global section",
+			configFile: "../testdata/cloudstack_config_missing_global_section.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "Invalid INI format",
+			configFile: "../testdata/cloudstack_config_invalid_format.ini",
+			wantErr:    true,
+		},
+		{
+			name:       "No instances",
+			configFile: "../testdata/cloudstack_config_no_instances.ini",
+			wantErr:    true,
+		},
+	}
 
-	tctx.restoreContext()
+	for _, tc := range tests {
+		t.Run(tc.name, func(tt *testing.T) {
+			var tctx testContext
+			tctx.backupContext()
+
+			g := NewWithT(t)
+			configString := test.ReadFile(t, tc.configFile)
+			encodedConfig := base64.StdEncoding.EncodeToString([]byte(configString))
+			os.Setenv(decoder.EksacloudStackCloudConfigB64SecretKey, encodedConfig)
```

Comment:
instead of os.Setenv, could you try using tt.Setenv, like we do in https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_webhook_test.go#L1225?

Reply:
Done